projects/components/src/list-view/list-view.component.ts

```diff
@@ -31,6 +33,9 @@ export class ListViewComponent {
 
   @Input()
   public records?: ListViewRecord[];
+
+  @Input()
+  public action?: TemplateRef<unknown>;
```

Comment:
We're getting close. This should work, but my suspicion is we'd get a lot more flexibility out of making this renderer the whole value renderer rather than always printing out the value as plaintext the following it with an optional additional renderer. A full value renderer can handle any use case this could, but also could do things like making the actual value itself the link - and IMO have a much more intuitive api.

The difference is code like:
```html
        <ht-list-view [records]="tagRecords" [action]="tagAction" data-sensitive-pii></ht-list-view>
        <ng-template #tagAction let-tag>
          <ht-explore-filter-link
            *ngIf="this.getExploreNavigationParams(tag) | async as exploreNavigationParams"
            [paramsOrUrl]="exploreNavigationParams"
            htTooltip="See traces in Explorer"
          >
          </ht-explore-filter-link
        ></ng-template>
```
vs code like (this would render a bit differently, depends exactly where we want value - but it shows the flexibility).
```html
        <ht-list-view [records]="tagRecords" data-sensitive-pii>
           <div *htListViewValueRenderer="let value">
              <ht-explore-filter-link 
                *ngIf="this.getExploreNavigationParams(tag) | async as exploreNavigationParams"
                [paramsOrUrl]="exploreNavigationParams"
                htTooltip="See traces in Explorer">
                {{ value }}
             </ht-explore-filter-link>
           </div>
       </ht-list-view>

Reply:
So, this `action` feature, is on a row level, not on the `value` level. So this design of having `listviewvaluerenderer`, I am not so sure about this. `tagRecords` already contains the value. This would have made more impact if we were going to have, different type of renderers for the both the columns (for key and value) like we have for the table. As far as I understand, list view has a sole purpose of showing the text value, right?

Reply:
So the whole purpose of this change is to change the renderer for the value column to go from just plain text to show a link after the plain text, right? That's basically a custom renderer for the value column, but we've special cased it to assume any custom renderer would want to be in addition to the plain text renderer, which may solves this use case but becomes redundant if we ever have to solve the general case. Imagine I just wanted to pretty print the values rather than using them as is (that is, format numbers, dates, etc.):

I could do something like
```html
        <ht-list-view [records]="tagRecords" data-sensitive-pii>
           <div *htListViewValueRenderer="let value">
               {{ value | prettyPrint }}
           </div>
       </ht-list-view>
```

> So, this action feature, is on a row level, not on the value level. So this design of having listviewvaluerenderer, I am not so sure about this. tagRecords already contains the value.

Either one kind of works the same. In either case, the list view takes in a template, instantiates it multiple times, and each time gives it some data. We can choose to give both the key and value, or just the value. I started with just the value as the implicit binding in my example, since that matched what you did with the action renderer and is extendable later with no changes to the api ( (e.g. `let-key="key" let-value="value" let-value-again`) 

> This would have made more impact if we were going to have, different type of renderers for the both the columns (for key and value) like we have for the table. As far as I understand, list view has a sole purpose of showing the text value, right?

I left the key renderer out of this because we didn't need it, but we could eventually add it. So, given the new requirement being added here to customize the values being rendered, the main differences with table are really around 1. interaction - paging, sorting, that kind of stuff and 2. having exactly two, known columns. So with table, we configure the columns and part of that is specifying a renderer for each. Here, we can shortcut that since we only need to (optionally) pass in a renderer.

This is how a lot of the library components out there work - with some variation depending on use case. For material table, a cell is defined in a similar way - the main difference is that there's a generic cellDef directive instead of a specific one which is associated with a column not by the directive being used, but by its location in a column def. Since we don't need headers or column defs, ours simplifies to just two named cell defs (only one of which we need right now).
```html
  <ng-container matColumnDef="position">
    <th mat-header-cell *matHeaderCellDef> No. </th>
    <td mat-cell *matCellDef="let element"> {{element.position}} </td>
  </ng-container>
```